flink-connector-test-util/src/main/java/com/ververica/cdc/connectors/utils/AssertUtils.java

```diff
@@ -0,0 +1,223 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ververica.cdc.connectors.utils;
+
+import io.debezium.data.Envelope;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.source.SourceRecord;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+/** Utilities for asserting {@link SourceRecord}. */
+public class AssertUtils {
+    /**
+     * Verify that the given {@link SourceRecord} is a {@link Envelope.Operation#CREATE
+     * INSERT/CREATE} record.
+     *
+     * @param record the source record; may not be null
+     */
+    public static void assertInsert(SourceRecord record, boolean keyExpected) {
+        if (keyExpected) {
+            assertNotNull(record.key());
+            assertNotNull(record.keySchema());
+        } else {
+            assertNull(record.key());
+            assertNull(record.keySchema());
+        }
+
+        assertNotNull(record.valueSchema());
+        Struct value = (Struct) record.value();
+        assertNotNull(value);
+        assertEquals(
+                Envelope.Operation.CREATE.code(), value.getString(Envelope.FieldName.OPERATION));
+        assertNotNull(value.get(Envelope.FieldName.AFTER));
+        assertNull(value.get(Envelope.FieldName.BEFORE));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 38:52, 100:113.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.